frontend/src/concepts/topology/PipelineDefaultTaskGroup.tsx

```diff
@@ -0,0 +1,110 @@
+import * as React from 'react';
+
+import {
+  LabelPosition,
+  WithSelectionProps,
+  isNode,
+  DefaultTaskGroup,
+  observer,
+  Node,
+  GraphElement,
+  RunStatus,
+  DEFAULT_LAYER,
+  Layer,
+  ScaleDetailsLevel,
+  TOP_LAYER,
+  NodeModel,
+  useHover,
+  PipelineNodeModel,
+} from '@patternfly/react-topology';
+import { Flex, FlexItem, Popover, Stack, StackItem } from '@patternfly/react-core';
+import { PipelineNodeModelExpanded, StandardTaskNodeData } from '~/concepts/topology/types';
+import NodeStatusIcon from '~/concepts/topology/NodeStatusIcon';
+import { NODE_HEIGHT, NODE_WIDTH } from './const';
+
+type PipelinesDefaultGroupProps = {
+  element: GraphElement<PipelineNodeModelExpanded>;
+} & WithSelectionProps;
+
+type PipelinesDefaultGroupInnerProps = Omit<PipelinesDefaultGroupProps, 'element'> & {
+  element: Node<PipelineNodeModel, StandardTaskNodeData>;
+};
+
+const DefaultTaskGroupInner: React.FunctionComponent<PipelinesDefaultGroupInnerProps> = observer(
+  ({ element, selected, onSelect }) => {
+    const [hover, hoverRef] = useHover();
+    const popoverRef = React.useRef<SVGGElement>(null);
+    const detailsLevel = element.getGraph().getDetailsLevel();
+
+    const getPopoverTasksList = (items: Node<NodeModel>[]) => (
+      <Stack hasGutter>
+        {items.map((item: Node) => (
+          <StackItem key={item.getId()}>
+            <Flex gap={{ default: 'gapXs' }}>
+              <FlexItem>
+                <NodeStatusIcon runStatus={item.getData()?.runStatus} />
```

Comment:
this icon is showed for runs and non runs. it makes sense for runs, but it will default to pending for pipelines or scheduled runs when there should just be no icon

Reply:
this is also true if the task is canceld or skipped then the underlaying tasks will be forever pending

Reply:
Thank you @Gkrumbach07 , good point! I've updated the popover content logic to show the status icons for runs only. Does this change also address your second comment here, or are you saying there some conditions we would have to update in the `NodeStatusIcon`?

Reply:
Yeah, for when a group node is skipped or canceled, then the ones bellow it will forever be pending. maybe make the default icon blank?